src/server/nuclearnet/web_socket_proxy_nuclearnet_server.ts

```diff
@@ -163,20 +162,29 @@ class PacketProcessor {
 
   private sendUnreliablePacket(event: string, packet: NUClearNetPacket) {
     // Throttle unreliable packets so that we do not overwhelm the client with traffic.
-    const done = this.enqueue(event)
+    const done = this.enqueue(event, this.clock.performanceNow())
     this.socket.send(event, packet, done)
-    this.clock.setTimeout(done, this.timeout)
+    this.clock.setTimeout(done.bind(this.clock.performanceNow() + this.timeout), this.timeout)
   }
 
-  private enqueue(event: string): () => void {
+  private enqueue(event: string, start: number): (time: number) => void {
     let isDone = false
-    const eventQueueSize = this.eventQueueSize.get(event) || 0
-    this.eventQueueSize.set(event, eventQueueSize + 1)
+    const tracking = this.eventTracking.get(event) || { active: 0, latency: 1.0, processing: 0.5 }
+    tracking.active += 1
+    this.eventTracking.set(event, tracking)
 
-    return () => {
+    return (processing: number) => {
       if (!isDone) {
-        const eventQueueSize = this.eventQueueSize.get(event) || 0
-        this.eventQueueSize.set(event, eventQueueSize - 1)
+        // Calculate network latency
+        const latency = this.clock.performanceNow() - start
+
+        // Update our performance tracking information
+        const tracking = this.eventTracking.get(event) || { active: 0, latency: 1.0, processing: 0.5 }
```

Comment:
This default shouldn't be here as it should be impossible to get here without having sent a packet

Reply:
You can use `this.eventTracking.get(event)!` in that case